ramos/mixins.py

```diff
@@ -24,14 +24,30 @@ class SingletonCreateMixin:
     _instances = {}
 
     @classmethod
-    def create(cls):
+    def _get_instance_key(cls, *args, **kwargs):
+        class_path = f'{cls.__module__}.{cls.__qualname__}'
+
+        args_key = '-'.join(str(arg) for arg in args) if args else ''
+
+        kwargs_key = (
+            '-'.join(f'{key}:{kwargs[key]}' for key in sorted(kwargs))
+            if kwargs
+            else ''
+        )
+
+        return f'{class_path}|{args_key}|{kwargs_key}'
```

Comment:
I think it would be better to call `hash` on arguments instead of `str`. There's no guarantee that the `__str__` will return an unique argument depending on the instance's relevant values and it doesn't need to be coherent with `__eq__`. The `__hash__` implementations should follow those rules.

you can use a tuples, instead of concatenating parameters:
```suggestion
        kwargs_tuple = tuple(sorted(kwargs.items()))

        return f'{class_path}|{hash(args)}|{hash(kwargs_tuple)}'
```

Reply:
Yes, looks really better. I changed the method as suggested.